services-directions-refresh-models/src/main/java/com/mapbox/api/directionsrefresh/v1/models/DirectionsRouteRefresh.java

```diff
@@ -0,0 +1,89 @@
+package com.mapbox.api.directionsrefresh.v1.models;
+
+import androidx.annotation.Nullable;
+import com.google.auto.value.AutoValue;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.TypeAdapter;
+import com.mapbox.api.directions.v5.models.DirectionsWaypoint;
+import com.mapbox.api.directionsrefresh.v1.DirectionsRefreshAdapterFactory;
+import java.util.List;
+
+/**
+ * Detailed information about an individual route such as the duration, distance and geometry.
+ */
+@AutoValue
+public abstract class DirectionsRouteRefresh extends DirectionsRefreshJsonObject {
+
+  /**
+   * Create a new instance of this class by using the {@link Builder} class.
+   *
+   * @return this classes {@link Builder} for creating a new instance
+   */
+  public static Builder builder() {
+    return new AutoValue_DirectionsRouteRefresh.Builder();
+  }
+
+  /**
+   * A Leg Refresh is an object contain refresh data between only two two {@link DirectionsWaypoint}.
+   *
+   * @return list of {@link RouteLegRefresh} objects
+   */
+  @Nullable
+  public abstract List<RouteLegRefresh> legs();
```

Comment:
Should this be a list? Isn't refresh API returning a value only for one leg?
/cc @danpat @danpaz

Reply:
No, it's a list. You could run https://github.com/mapbox/mapbox-java/blob/master/samples/src/main/java/com/mapbox/samples/BasicDirectionsRefresh.java and ger instance of response

Reply:
I see, but the list will always have only one element since we're defining the leg index as an argument, is that right?

Reply:
param `leg index` means "current leg index", so the response contains annotations for current and following legs(annotations for passed legs aren't present)

Reply:
gotcha, thanks :+1: